Signal/src/AppDelegate.m

```diff
@@ -219,13 +221,44 @@ - (void)ensureIsReadyForAppExtensions
         return;
     }
 
+    NSError *_Nullable error = [self convertDatabaseIfNecessary];
+    // TODO: Handle this error.
+    OWSAssert(!error);
+
     [NSUserDefaults migrateToSharedUserDefaults];
 
     [TSStorageManager migrateToSharedData];
     [OWSProfileManager migrateToSharedData];
     [TSAttachmentStream migrateToSharedData];
 }
 
+- (nullable NSError *)convertDatabaseIfNecessary
+{
+    NSString *databaseFilePath = [TSStorageManager legacyDatabaseFilePath];
+
+    NSError *error;
+    NSData *_Nullable databasePassword = [OWSStorage tryToLoadDatabasePassword:&error];
+    if (!databasePassword || error) {
```

Comment:
There's a card to handle these errors (by presumably presenting a UI that allows users to report these failures).  Let's sort this out in that PR.

Reply:
I'm not sure I made myself clear...

I'm not saying *if something goes wrong* we should handle it, i'm saying won't this fail *every* time upon installing to a new device, because the password doesn't exist in the keychain yet?

Reply:
Yup, I think you're probably right. I'm just saying let's straighten it out when we do that follow up work since this PR is merged.

Reply:
Ah got it. I'm going to address this in the work I'm doing now.